### PR TITLE
More robust guest members for users and tests for it

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
More robust guest members for users and tests for it

#### Issues
- https://mattermost.atlassian.net/browse/MM-17237
- https://mattermost.atlassian.net/browse/MM-17223
- https://mattermost.atlassian.net/browse/MM-17222
- maybe https://mattermost.atlassian.net/browse/MM-17093

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1939,7 +1939,7 @@ func (s SqlChannelStore) AnalyticsDeletedTypeCount(teamId string, channelType st
 
 func (s SqlChannelStore) GetMembersForUser(teamId string, userId string) (*model.ChannelMembers, *model.AppError) {
 	var dbMembers channelMemberWithSchemeRolesList
-	_, err := s.GetReplica().Select(&dbMembers, CHANNEL_MEMBERS_WITH_SCHEME_SELECT_QUERY+"WHERE ChannelMembers.UserId = :UserId AND (Teams.Id = :TeamId OR Teams.Id IS NULL)", map[string]interface{}{"TeamId": teamId, "UserId": userId})
+	_, err := s.GetReplica().Select(&dbMembers, CHANNEL_MEMBERS_WITH_SCHEME_SELECT_QUERY+"WHERE ChannelMembers.UserId = :UserId AND (Teams.Id = :TeamId OR Teams.Id = '' OR Teams.Id IS NULL)", map[string]interface{}{"TeamId": teamId, "UserId": userId})
```
Comment (gabrieljackson) [inline store/sqlstore/channel_store.go]: Jumping in with a quick question to see what other open tickets this change may affect.

In what cases are we seeing that `Teams.Id = ''`?

Comment (jespino) [reply store/sqlstore/channel_store.go]: Team.Id = '' means DirectChannel or GroupChannel, anything related to getting the memberships of a team or memberships of directchannels or group channels can be affected by this.

Comment (gabrieljackson) [reply store/sqlstore/channel_store.go]: Okay, that totally checks out. Thanks for explaining.

**Comment by srkgupta** [comment]
QA Tests updated based on the ticket https://mattermost.atlassian.net/browse/MM-17222

**Pull request merged**